ledger/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 #![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(min_specialization))]
 #![allow(clippy::arithmetic_side_effects)]
+#![recursion_limit = "2048"]
```

Comment:
This was just a precaution right; we didn't actually bump up against this / this would cause compile time break right ?

Reply:
It's real, without it `cargo +nightly-2024-01-05` throws this error:
```
error: recursion limit reached while expanding `eager_internal!`
    --> ledger/src/blockstore_processor.rs:1153:9
     |
1153 | /         lazy! {
1154 | |             datapoint_info!(
1155 | |                 "replay-slot-end-to-end-stats",
1156 | |                 ("slot", slot as i64, i64),
...    |
1162 | |             );
1163 | |         };
```

Reply:
Also now that this code is out of program-runtime, program-runtime now compiles if I remove the `#![recursion_limit = "2048"]` in its lib.rs so I'm removing it from there